test/assignment/test.range.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+const range = require('../../lib/assignment/range');
+const _ = require('lodash');
+
+describe('Range Assignment', function () {
+  const topicPartition = {
+    'RebalanceTopic': [
+      '0',
+      '1',
+      '2'
+    ],
+    'RebalanceTest': [
+      '0',
+      '1',
+      '2'
+    ]
+  };
+
+  const groupMembers = [
+    {
+      'subscription': [
+        'RebalanceTopic',
+        'RebalanceTest'
+      ],
+      'version': 0,
+      'id': 'consumer1'
+    },
+    {
+      'subscription': [
+        'RebalanceTopic',
+        'RebalanceTest'
+      ],
+      'version': 0,
+      'id': 'consumer2'
+    }
+  ];
+
+  it('should have required fields', function () {
+    range.assign.should.be.function;
+    range.name.should.be.eql('range');
+    range.version.should.be.eql(0);
+  });
+
+  it('should partition two topics of three partitions between two consumers', function (done) {
+    range.assign(topicPartition, groupMembers, function (error, result) {
+      const consumer1 = _.first(result);
+      consumer1.memberId.should.eql('consumer1');
+      Object.keys(consumer1.topicPartitions).should.eql(['RebalanceTopic', 'RebalanceTest']);
+      consumer1.topicPartitions['RebalanceTest'].should.eql([0, 1]);
+      consumer1.topicPartitions['RebalanceTopic'].should.eql([0, 1]);
+
+      const consumer2 = _.last(result);
+      consumer2.memberId.should.eql('consumer2');
+      Object.keys(consumer2.topicPartitions).should.eql(['RebalanceTopic', 'RebalanceTest']);
+      consumer2.topicPartitions['RebalanceTest'].should.eql([2]);
+      consumer2.topicPartitions['RebalanceTopic'].should.eql([2]);
+
+      done(error);
```

Comment:
Should you explicitly check the error earlier in the test to make sure the test failure message you see is the most helpful one? (Same for other tests...)